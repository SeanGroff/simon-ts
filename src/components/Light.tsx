import * as React from 'react';
import styled, { keyframes, StyledProps } from 'styled-components';

interface Props {
  id: number;
  animate: boolean;
  color: string;
  lightUpColor: string;
  clickable: boolean;
  topLeft?: boolean;
  topRight?: boolean;
  bottomLeft?: boolean;
  bottomRight?: boolean;
  onClick(): void;
}

interface LightUpProps {
  color: string;
  lightUpColor: string;
}

const Light: React.StatelessComponent<Props> = props => {
  console.log(props);
  return <div />;
};

const LightUp = (props: LightUpProps) => keyframes`
  0% {
    background-color: ${props.color};;
  }
  30%, 60% {
    background-color: ${props.lightUpColor};
  }
  100% {
    background-color: ${props.color};
  }
`;

const StyledLight = styled(Light)`
  display: inline-block;
  width: 200px;
  height: 200px;
  background-color: ${(props: StyledProps<Props>) => props.color};
  border: 12px solid #333333;
  border-top-left-radius: ${(props: StyledProps<Props>) =>
    props.topLeft ? '100%' : '0'};
  border-top-right-radius: ${(props: StyledProps<Props>) =>
    props.topRight ? '100%' : '0'};
  border-bottom-left-radius: ${(props: StyledProps<Props>) =>
    props.bottomLeft ? '100%' : '0'};
  border-bottom-right-radius: ${(props: StyledProps<Props>) =>
    props.bottomRight ? '100%' : '0'};
  pointer-events: ${(props: StyledProps<Props>) =>
    props.clickable ? 'auto' : 'none'};
  animation: ${(props: StyledProps<Props>) =>
    props.animate ? `${LightUp(props)} 1500ms linear` : 'none'};
  :hover {
    cursor: ${(props: StyledProps<Props>) =>
      props.clickable ? 'pointer' : 'initial'};
  }
`;

export default (props: Props) => <StyledLight {...props} />;
