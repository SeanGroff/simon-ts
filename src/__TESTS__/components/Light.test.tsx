global.requestAnimationFrame = callback => {
  setTimeout(callback, 0);
};

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import Light from '../../components/Light';

configure({ adapter: new Adapter() });

it('should render a Light with a green background', () => {
  const component = shallow(<Light color="green" />);
  expect(component).toMatchSnapshot();
});
