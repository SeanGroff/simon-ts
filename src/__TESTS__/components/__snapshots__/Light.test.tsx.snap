// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Light with a green background 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    color="green"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "color": "green",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "color": "green",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
